### ISSUE TITLE
[PyTorch] Error "could not create a primitive"

### ISSUE THREAD

**Issue by Aidyn-A** [oneapi-src/oneDNN#1500]
# Summary
A simple convolution fails in https://github.com/pytorch/pytorch/commit/7ad87f63e248b629d435a199cb61f4ed1f3dfcab with a "could not create a primitive" error on AMD EPYC 7742 CPU.

The minimal code snippet is:
```python
import torch

x = torch.rand(1, 32, 512, 512, 256)
m = torch.nn.Conv3d(32, 1, kernel_size=1, padding=0, stride=1, bias=False)
y = m(x)
```

Its output:
```
  File "test.py", line 5, in <module>
    y = m(x)
  File "/opt/pytorch/pytorch/torch/nn/modules/module.py", line 1423, in _call_impl
    return forward_call(*input, **kwargs)
  File "/opt/pytorch/pytorch/torch/nn/modules/conv.py", line 613, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/opt/pytorch/pytorch/torch/nn/modules/conv.py", line 608, in _conv_forward
    return F.conv3d(
RuntimeError: could not create a primitive
```

# Version
oneDNN v2.7 (650085b2f3643aad05c629425983491d63b5c289)
https://github.com/pytorch/pytorch/commit/7ad87f63e248b629d435a199cb61f4ed1f3dfcab and newer

# Environment
```
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   43 bits physical, 48 bits virtual
CPU(s):                          256
On-line CPU(s) list:             0-255
Thread(s) per core:              2
Core(s) per socket:              64
Socket(s):                       2
NUMA node(s):                    8
Vendor ID:                       AuthenticAMD
CPU family:                      23
Model:                           49
Model name:                      AMD EPYC 7742 64-Core Processor
Stepping:                        0
Frequency boost:                 enabled
CPU MHz:                         2250.000
CPU max MHz:                     2250.0000
CPU min MHz:                     1500.0000
BogoMIPS:                        4491.83
Virtualization:                  AMD-V
L1d cache:                       4 MiB
L1i cache:                       4 MiB
L2 cache:                        64 MiB
L3 cache:                        512 MiB
NUMA node0 CPU(s):               0-15,128-143
NUMA node1 CPU(s):               16-31,144-159
NUMA node2 CPU(s):               32-47,160-175
NUMA node3 CPU(s):               48-63,176-191
NUMA node4 CPU(s):               64-79,192-207
NUMA node5 CPU(s):               80-95,208-223
NUMA node6 CPU(s):               96-111,224-239
NUMA node7 CPU(s):               112-127,240-255
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP conditional, RSB filling
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_goo
                                 d nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_lega
                                 cy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_ps
                                 tate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_l
                                 lc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pft
                                 hreshold avic v_vmsave_vmload vgif v_spec_ctrl umip rdpid overflow_recov succor smca sme sev sev_es
```

**Comment by vpirogov** [comment]
Thanks for the report, Aidyn-A. As your issue is reproducible in Pytorch I'd suggest opening an issue at [Pytorch Github](https://github.com/pytorch/pytorch) first.

+@aice-support, @XiaobingSuper, @jgong5, @gujinghui

**Comment by yehudaorel** [comment]
@vpirogov I was able to reproduce this issue on 2nd gen AMD processor.

**Comment by vpirogov** [comment]
Thanks, @yehudaorel. Does the system have enough memory to handle the problem? I believe the activation size is 8 Gb, so the total memory would be north of 16 Gb. Can you reproduce it on smaller sizes?

**Comment by yehudaorel** [comment]
@vpirogov I used a AWS instance with 31Gb of memory, so it should be more than enough.
`x = torch.rand(1, 32, 362, 362, 256) ` Worked with no fails, but (363, 363)+ result in the same issue.

**Comment by vpirogov** [comment]
Thanks, @yehudaorel.

I can reproduce the issue with Intel AVX2 codepath (on an older Xeon), but not with Intel AVX512 codepath. So the is not AMD specific.
```
$ DNNL_VERBOSE=1 ./benchdnn --conv mb1ic32ih512iw512id256oc1kh1kw1kd1ph0pw0pd0
onednn_verbose,info,oneDNN v2.7.1 (commit 9e9b859fddc6f813f9b9cac093d7d131c84054ab)
onednn_verbose,info,cpu,runtime:OpenMP,nthr:44
onednn_verbose,info,cpu,isa:Intel AVX2
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
error [int create_primitive(benchdnn_dnnl_wrapper_t<dnnl_primitive*>&, dnnl_engine_t, const func_t&, const prb_t*, res_t*, dir_t, const_dnnl_primitive_desc_t, bool) [with func_t = dnnl_status_t(init_pd_args_t<conv::prb_t>&); prb_t = conv::prb_t; dnnl_engine_t = dnnl_engine*; const_dnnl_primitive_desc_t = const dnnl_primitive_desc*]:540]: 'dnnl_primitive_create(&prim, pdw)' -> runtime_error(5)
@@@ error [int init_prim(benchdnn_dnnl_wrapper_t<dnnl_primitive*>&, const func_t&, const prb_t*, res_t*, dir_t, const_dnnl_primitive_desc_t, bool) [with func_t = dnnl_status_t(init_pd_args_t<conv::prb_t>&); prb_t = conv::prb_t; const_dnnl_primitive_desc_t = const dnnl_primitive_desc*]:585]: 'create_primitive(primw, engine, init_pd_func, prb, res, dir, hint, is_service_prim)' -> 1
@@@ error [int conv::doit(const conv::prb_t*, res_t*):501]: 'init_prim(prim, init_pd, prb, res)' -> 1
0:UNTESTED_FAILED __REPRO: --conv mb1ic32id256ih512iw512oc1od256oh512ow512kd1kh1kw1pd0ph0pw0
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:1 listed:0
total compute_ref: sum(s):0.00
```

**Comment by vpirogov** [comment]
Fixed in [oneDNN v2.7.2](https://github.com/oneapi-src/oneDNN/releases/tag/v2.7.2).
